core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -766,9 +782,11 @@ private[spark] class TaskSetManager(
         logWarning(failureReason)
         None
     }
-    // always add to failed executors
-    failedExecutors.getOrElseUpdate(index, new HashMap[String, Long]()).
-      put(info.executorId, clock.getTimeMillis())
+
+    if (reason.countTowardsTaskFailures) {
```

Comment:
`failedExecutors` was the variable for the old blacklisting mechanism, its completely gone now.  had nothing to do with executors that have failed completely, despite the name:

```scala
// key is taskId (aka TaskInfo.index), value is a Map of executor id to when it failed
private val failedExecutors = new HashMap[Int, HashMap[String, Long]]()
```